### PR TITLE
Development

### PR DESCRIPTION
**Pull request by chernikovanna**
I think we're done moving over most of the functions in hl.c to ops.c/ops.h and having those get built into the squashfuse library

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -27,13 +27,13 @@ pkgconfig_DATA 	= squashfuse.pc
...
-	nonstd-pread.c nonstd-stat.c \
+	nonstd-pread.c nonstd-stat.c ops.c \
 	squashfs_fs.h common.h nonstd-internal.h nonstd.h swap.h cache.h table.h \
 	dir.h file.h decompress.h xattr.h squashfuse.h hash.h stack.h traverse.h \
-	util.h fs.h
+	util.h fs.h ops.h
 libsquashfuse_la_CPPFLAGS = $(ZLIB_CPPFLAGS) $(XZ_CPPFLAGS) $(LZO_CPPFLAGS) \
 	$(LZ4_CPPFLAGS) $(ZSTD_CPPFLAGS)
-libsquashfuse_la_LIBADD =
+libsquashfuse_la_LIBADD = libfuseprivate.la $(FUSE_LIBS)
```
Comment (j-ogas) [inline Makefile.am]: What does this do? I am able to build and link against your fork without this line. I wonder if it is necessary.

File: ops.c [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+                if (err)
+                        return err;
+                if (!found)
+                        return SQFS_ERR;
+        }
+        return SQFS_OK;
+}
+
+
+
```
Comment (j-ogas) [inline ops.c]: Remove spurious new line.
```suggestion
```

File: ops.c [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        sqfs_inode inode;
+        if (sqfs_hl_lookup(&fs, &inode, path))
+                return -ENOENT;
+
+        if (sqfs_stat(fs, &inode, st))
+                return -ENOENT;
+
+        return 0;
+}
+ int sqfs_hl_op_opendir(const char *path, struct fuse_file_info *fi) {
```
Comment (j-ogas) [inline ops.c]: Add a space between functions.
```suggestion
}

 int sqfs_hl_op_opendir(const char *path, struct fuse_file_info *fi) {
```

File: ops.c [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+
+        if ((sqfs_xattr_lookup(fs, &inode, name, value, &real)))
+                return -EIO;
+        if (real == 0)
+                return -sqfs_enoattr();
+        if (size != 0 && size < real)
+                return -ERANGE;
+        return real;
+}
+/** returns fuse_operations struct populated with function pointers to above operation functions*/
```
Comment (j-ogas) [inline ops.c]: Insert space between functions.

Also, each comment line should be 80 characters max. Break the comment into two separate lines.
```suggestion
}

/** returns fuse_operations struct populated with function pointers to above operation functions*/
```

**Review by j-ogas — CHANGES_REQUESTED** [review]
Good work 👍. See requested changes and comments below.

File: Makefile.am (hunk 1 above)
Comment (chernikovanna) [reply Makefile.am]: We could try to build and link without it! This may have been something we added at a time when we weren't linking correctly and thought we needed it ':D

**Review by atorrez — CHANGES_REQUESTED** [review]
I think this looks good overall.  I think it would be good to have a general description of ops.c at the top of the file in a comment block.  We can get consensus on this though.  Code functionality is not hard to discern but wondering if some comments are needed?

File: Makefile.am (hunk 1 above)
Comment (chernikovanna) [reply Makefile.am]: So I'm noticing that I'm able to build and link to the generated static libraries, and run our code that depends on the libraries. but there are issues when trying to build the .so libraries in squashfuse, saying that there's an undefined reference to functions located in libfuseprivate. I imagine this is because the ops code was originally in the source file which linked both libfuseprivate and libsquashfuse. 
TLDR; I think for our purposes we don't need that line of code, but for squashfuse to function all around I think we do.

**Review by shanegoff — APPROVED** [review]
The pull request submitted looks very good.
